x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -864,6 +871,32 @@ function getCorrespondingAction(
     ) as RawRuleAction;
   }
 }
+/**
+ * removes internal tags(starts with '__internal') from Security Solution rules
+ * @param doc rule to be migrated
+ * @returns migrated rule if it's Security Solution rule or unchanged if not
+ */
+function removeInternalTags(
+  doc: SavedObjectUnsanitizedDoc<RawRule>
+): SavedObjectUnsanitizedDoc<RawRule> {
+  if (!isDetectionEngineAADRuleType(doc)) {
+    return doc;
+  }
+
+  const {
+    attributes: { tags },
+  } = doc;
+
+  const filteredTags = tags.filter((tag) => !tag.startsWith('__internal_'));
```

Comment:
Is it possible for tags to be undefined? Perhaps we should be safe and add a fallback `(tags ?? []).filter`?

Reply:
@ymao1 I don't know, they defined as `string[]` and usually accessed across app without additional checks
Here is, its typings https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/types.ts#L229
I can see, it's defined within alerting framework. 
Is these typings are safe to use? Should we all the time check if field is not `undefined` before doing any actions on it?

Meanwhile, I've added check to not block this PR and we can be sure it's safe